trace/google/cloud/trace/_gax.py

```diff
@@ -0,0 +1,197 @@
+            filter_=None,
+            order_by=None,
+            page_token=None):
+        """Returns of a list of traces that match the specified filter
+        conditions.
+
+        :type project_id: str
+        :param project_id: ID of the Cloud project where the trace data is
+                           stored.
+
+        :type view: :class:`google.cloud.gapic.trace.v1.enums.
+                           ListTracesRequest.ViewType`
+        :param view: (Optional) Type of data returned for traces in the list.
+                     Default is ``MINIMAL``.
+
+        :type page_size: int
+        :param page_size: (Optional) Maximum number of traces to return.
+                          If not specified or <= 0, the implementation selects
+                          a reasonable value. The implementation may return
+                          fewer traces than the requested page size.
+
+        :type start_time: :class:`google.protobuf.timestamp_pb2.Timestamp`
+        :param start_time: (Optional) Start of the time interval (inclusive)
+                           during which the trace data was collected from the
+                           application.
+
+        :type end_time: :class:`google.protobuf.timestamp_pb2.Timestamp`
+        :param end_time: (Optional) End of the time interval (inclusive)
+                         during which the trace data was collected from the
+                         application.
+
+        :type filter: str
+        :param filter_: (Optional) An optional filter for the request.
+
+        :type order_by: str
+        :param order_by: (Optional) Field used to sort the returned traces.
+
+        :type page_token: str
+        :param page_token: opaque marker for the next "page" of entries. If not
+                           passed, the API will return the first page of
+                           entries.
+
+        :rtype: dict
+        :returns: Traces that match the specified filter conditions.
+        """
+        if page_token is None:
+            page_token = INITIAL_PAGE
+        options = CallOptions(page_token=page_token)
+        page_iter = self._gax_api.list_traces(
+            project_id=project_id,
+            view=view,
+            page_size=page_size,
+            start_time=start_time,
+            end_time=end_time,
+            filter_=filter_,
+            order_by=order_by,
+            options=options)
+        item_to_value = _item_to_mapping
+        return GAXIterator(self.client, page_iter, item_to_value)
+
+
+def _parse_trace_pb(trace_pb):
+    """Parse a ``Trace`` protobuf to a dictionary.
+
+    :type trace_pb: :class:`google.cloud.proto.devtools.cloudtrace.v1.
+                            trace_pb2.Trace`
+    :param trace_pb: A trace protobuf instance.
+
+    :rtype: dict
+    :returns: The converted trace dict.
+    """
+    try:
+        return MessageToDict(trace_pb)
+    except TypeError:
+        raise
+
+
+def _item_to_mapping(iterator, trace_pb):
```

Comment:
If we're not going to do an HTTP version, I think it's probably preferable to return a proto instead of a dict from the interator. @dhermes @lukesneeringer to confirm.